### ISSUE TITLE
ipv6 format between systemd and sslh

### ISSUE THREAD

**Issue by seblu** [yrutschle/sslh#104]
Hello,

_sslh.cfg_ and systemd-sslh-generator use incompatible IPv6 address format. systemd require IPv6 surrounded by **[]** when sslh.cfg refuse them.
### So with the following config,

```
listen:
(
    { host: "::0"; port: "443"; }
);
```
### generated unit is refused by systemd

> sept. 28 01:00:15 mars.seblu.net systemd[1]: [/run/systemd/generator/sslh.socket:10] Failed to parse address value, ignoring: ::0:443
> sept. 28 01:00:15 mars.seblu.net systemd[1]: sslh.socket: Unit lacks Listen setting. Refusing.
### and with

```
listen:
(
    { host: "[::0]"; port: "443"; }
);
```
### sslh refuse it

> sept. 28 01:09:14 mars.seblu.net sslh[14838]: /etc/sslh/sslh.cfg:file I/O error
> sept. 28 01:09:15 mars.seblu.net sslh[14838]: Name or service not known `[::0]:443'

**Comment by dlsmartins** [comment]
Just my 50 cents: to what host, exactly, are you trying to bind?
In IPv6, lo is [::1] and the 0.0.0.0 equivalent is [::]. I'm not sure of what [::0] resolves to.
Hope it helps :)

**Comment by seblu** [comment]
`[[::]]` and `[[::0]]` both mean ipv6 `0000:0000:0000:0000:0000:0000:0000:0000`

sslh doesn't like **[]** notation whatever you put inside.
Here is the output with your suggestion.

> nov. 03 19:12:00 sheldon systemd[1]: Started SSL/SSH multiplexer.
> nov. 03 19:12:00 sheldon sslh-fork[6481]: Name or service not known `[::]:443'
> nov. 03 19:12:00 sheldon sslh-fork[6481]: No listening address specified; use at least one -p option

**Comment by yrutschle** [comment]
I'm undecided between:
a- Extending to systemd-sslh-generator to guess if it's a IPv6 address and output it with brackets
b- Make sslh config ignore brackets in the IP field so you can add them to the config file

(a) is probably a little more complicated to code, but (b) makes for a non-standard writing of addresses in the config file (brackets are normally used for URIs, not single IPs, as far as I understand).

Are there arguments from the user side I'm missing?

**Comment by yrutschle** [comment]
Fixed in 00d5872aa176a23588f143551aa4b7e8dd1c0ca6. Funnily enough, the code already existed for the command line parsing.
